src/audio/component.c

```diff
@@ -29,23 +29,67 @@ DECLARE_SOF_UUID("component", comp_uuid, 0x7c42ce8b, 0x0108, 0x43d0,
 
 DECLARE_TR_CTX(comp_tr, SOF_UUID(comp_uuid), LOG_LEVEL_INFO);
 
-static const struct comp_driver *get_drv(uint32_t type)
+static const struct comp_driver *get_drv(struct sof_ipc_comp *comp)
 {
 	struct comp_driver_list *drivers = comp_drivers_get();
 	struct list_item *clist;
 	const struct comp_driver *drv = NULL;
 	struct comp_driver_info *info;
+	struct sof_ipc_comp_new_ext *comp_ext;
+
+	/* use UUID to match the driver if UUID is provided */
+	if (comp->ext_data_offset) {
+		/* validate the header size */
+		if (comp->hdr.size < comp->ext_data_offset +
+		    sizeof(struct sof_ipc_comp_new_ext)) {
```

Comment:
This doesnt allow ext data to grow. It may grow with some additional fileds before IPC2.0 is merged.